include/openssl/pkcs7.h

```diff
@@ -332,13 +415,45 @@ OPENSSL_EXPORT int PKCS7_type_is_signedAndEnveloped(const PKCS7 *p7);
 // must be NULL and |flags| must be exactly |PKCS7_NOATTR | PKCS7_BINARY |
 // PKCS7_NOCERTS | PKCS7_DETACHED|.
 //
+// The third mode is used for more general signing and does not require the
+// specification of any flags, but does require |sign_cert|, |pkey|, and |data|
+// to be populated. This mode always behaves as if |PKCS7_NOATTR| and
+// |PKCS7_BINARY| are set. It honors the specification (or elision) of
+// |PKCS7_DETACHED|. It does not allow |PKCS7_NOCERTS|. This mode is provided
+// primarily for compatibility with Ruby.
+//
 // Note this function only implements a subset of the corresponding OpenSSL
 // function. It is provided for backwards compatibility only.
```

Comment:
* The "compatibility with Ruby" sounds odd to me. We're improving our compatibility with OpenSSL for the purpose of supporting our integration with Ruby -- but I'm not sure how best to say that here.
* Now that we've added this third mode, do we still have only a "subset" of OpenSSL's functionality?

Reply:
> but I'm not sure how best to say that here.

yeah, agreed we can drop this sentence.

> do we still have only a "subset" of OpenSSL's functionality?

yes. we don't the various flags documented above on L400-L405. i think the "subset" statement will hold until we support all of those.